---
title: "analyze_genomics"
output: html_document
---

# Packages

Download packages you will be using for the analysis!

```{r, setup}
# install dplyr, readr, tidyr, ggplot2
install.packages(c("dplyr", "readr", "tidyr", "ggplot2"))

# load all your packages
library(dplyr)
library(readr)
library(tidyr)
library(ggplot2)

```

# Load data

Load the genomics data that you downloaded from https://github.com/AndersenLab/Computational-Bootcamp/blob/master/Data/subset_Soft_Filter_Training.tsv.gz

```{r, load_data}

# load data using function readr::read_tsv()
# don't know how to use read_tsv? TRY: ?read_tsv
genomics_data <- readr::read_tsv("~/20180726_bootcamp/data/raw/subset_Soft_Filter_Training.tsv.gz")

# View your data
View(genomics_data)

# load data again, using colnames = F
genomics_data <- readr::read_tsv("~/20180726_bootcamp/data/raw/subset_Soft_Filter_Training.tsv.gz", col_names = F)

# add a vector of column names using the function colnames()
colnames(genomics_data) <- c("chrom", "pos", "ref", "alt", "qual", "variant_filter", "allele_ct", "allele_freq", "strain", "genotype", "depth", "strain_filter", "allele_depth")

# inspect data
head(genomics_data)
str(genomics_data)

```

# Process data

Create a smaller dataframe to work with.

```{r, process_data}

# generate a smaller data set for faster processing
# use dplyr::filter() to keep only chromosomes IV and V
smaller_data <- dplyr::filter(genomics_data, chrom %in% c("IV","V"))

# Save smaller dataframe as an .Rda file using function save()
save(smaller_data, file = ("~/20180726_bootcamp/data/processed/smaller_data.Rda"))

# Remove genomics_data from workspace
rm(genomics_data)

```

# Analyze data

Analyze the data by counting the number of strains and removing some variants.

```{r, analyze_data}

# how many strains do we have in the data set?
# HINT: try using the functions: length() and unique()
length(unique(smaller_data$strain))

# remove variants that have high levels of heterozygosity
# HINT: look at the unique values of the variant_filter column
unique(smaller_data$variant_filter)

no_het_variants <- dplyr::filter(smaller_data, variant_filter != "high_heterozygosity")

# remove multi-allelic variants
# Multi-allelic variants can be identified by having a "," (e.g. C,T or AG,TC) in the 'alt' column
# HINT: try ?grepl() to learn how to match a pattern in a string instead of the whole string
no_comma <- dplyr::filter(no_het_variants, !grepl(",", alt))

# Use the pipe (%>%) to combine steps with dplyr!
biallelic_variants <- smaller_data %>%
    dplyr::filter(variant_filter != "high_heterozygosity") %>%
    dplyr::filter(!grepl(",", alt))

# Keep piping to add multiple steps!
# 1. filter to keep only strains that PASS the strain_filter
# 2. Look at the genotype column, filter out any irregularities
# 3. Make a new column (GT) that translates the genotype into "REF" or "ALT" using the function dplyr::mutate()
# 4. Remove the genotype column (try dplyr::select())
# 5. Group each variant by its genotype using the function dplyr::group_by()
###### Each variant has a unique 'chrom' and 'pos' and the genotype is designated in our new column 'GT'
# 6. Make a new column called 'num_geno' that adds the number of strains with REF or ALT using dplyr::summarize() 
###### HINT: look at ?n()
# 7. Spread the 'GT' column and fill with the 'num_geno' to create two columns (REF and ALT) whose values are the num_geno
###### HINT: look at tidyr::spread()
# 8. Make a new column called 'perc_ref' that calculates the percent of strains that have the REF genotype at each variant
biallelic_variants <- smaller_data %>%
    dplyr::filter(variant_filter != "high_heterozygosity") %>%
    dplyr::filter(!grepl(",", alt)) %>%
    dplyr::filter(strain_filter == "PASS") %>%
    dplyr::filter(genotype != "./.") %>%
    dplyr::mutate(GT = ifelse(genotype == "0/0", "REF", "ALT")) %>%
    dplyr::select(-genotype) %>%
    dplyr::group_by(chrom, pos, GT) %>%
    dplyr::summarize(num_geno = n()) %>%
    tidyr::spread(GT, num_geno, fill = 0) %>%
    dplyr::mutate(perc_ref = REF / (REF + ALT))

```


# Plot your data!

Use ggplot2 to make beautiful plots to show off all your hard work.

```{r, plot_data}
    
# Plot a histogram of your biallelic_variants dataframe with perc_ref on the x-axis
# HINT: use ggplot2::geom_histogram()
ggplot2::ggplot(data = biallelic_variants) +
    ggplot2::aes(x = perc_ref) +
    ggplot2::geom_histogram()

# Using the same histogram as above, now facet by chromosome using ggplot2::facet_grid()
# HINT: ~ means 'by'
ggplot2::ggplot(data = biallelic_variants) +
    ggplot2::aes(x = perc_ref) +
    ggplot2::geom_histogram() + 
    ggplot2::facet_grid(~chrom)

# Using the same histogram as above, try coloring the histogram by chromosome
# HINT: add color within the aesthetics (fill = variable)
ggplot2::ggplot(data = biallelic_variants) +
    ggplot2::aes(x = perc_ref, fill = chrom) +
    ggplot2::geom_histogram() + 
    ggplot2::facet_grid(~chrom)

# Assign your plot to a variable name
histogram <- ggplot2::ggplot(data = biallelic_variants) +
    ggplot2::aes(x = perc_ref, fill = chrom) +
    ggplot2::geom_histogram() + 
    ggplot2::facet_grid(~chrom)

# Save your plot using ggplot2::ggsave()
ggplot2::ggsave(histogram, "~/20180726_bootcamp/figures/histogram.png")

# OYO: Try making a scatterplot of perc_ref for each position across the genome (for each chromosome)
# Color all points red instead of black
scatterplot <- ggplot2::ggplot(data = biallelic_variants) +
    ggplot2::aes(x = pos, y = perc_ref) +
    ggplot2::geom_point(color = 'red') +
    ggplot2::facet_grid(~chrom)

# Save your plot using ggplot2::ggsave()
ggplot2::ggsave(scatterplot, "~/20180726_bootcamp/figures/scatterplot.png")

```


# On your own plotting/analysis

< Fill in what type of analysis you performed here as well as what your conclusions were from the data >

```{r, on_you_own}

# YOUR ANALYSIS HERE


# YOUR PLOT HERE


```

